runtime/src/tests/runtime.rs

```diff
@@ -307,6 +307,43 @@ fn test_toml_manifest() {
     assert_eq!(count.get("count").unwrap().as_i64().unwrap(), 1);
 }
 
+#[test]
+fn test_call_with_context() {
+    // assert!(set_log_file("stdout", Some(log::Level::Trace)));
+    #[derive(Clone)]
+    struct Foo {
+        message: String,
+    }
+
+    let f = Function::new(
+        "host_reflect",
+        [PTR],
+        [PTR],
+        UserData::default(),
+        |current_plugin, _val, ret, _user_data: UserData<()>| {
+            let foo = current_plugin.context::<Foo>()?;
+            let hnd = current_plugin.memory_new(foo.message)?;
+            ret[0] = current_plugin.memory_to_val(hnd);
+            Ok(())
+        },
+    );
+
+    let mut plugin = Plugin::new(WASM_REFLECT, [f], true).unwrap();
+
+    let message = "hello world";
+    let output: String = plugin
+        .call_with_context(
+            "reflect",
+            "anything, really",
```

Comment:
hah, this was my initial thought - "then what's the input?" but I realized that the `context` value is only available in host functions (assuming we forbid access to the actual global somehow), which feels like a pretty powerful combination. this seems like a nice stepping-stone to allowing externref arguments while compilers are still catching up!

there could be some confusion around what to pass as input vs context (especially since we used to have a `Context` type that meant something totally different)

Reply:
Maybe something like `HostContext` or `HostParam` might be more clear - open to other ideas too!

Reply:
Good point – I renamed to `host_context` / `HostContext` in the applicable places.